File: spoj/3375-STAMPS/solution_3375.cpp
#include <bits/stdc++.h>
using namespace std;

int main()
{
    int tests;
    cin >> tests;
    for (int t = 0; t < tests; t++)
    {
        int rayStamps, n;
        cin >> rayStamps >> n;

        int friendStamps[n];

        for (int i = 0; i < n; i++)
            cin >> friendStamps[i];

        sort(friendStamps, friendStamps + n);

        int i = n - 1;
        int count = 0;
        while (i >= 0 && rayStamps > 0)
        {
            count++;
            rayStamps -= friendStamps[i];
            i--;
        }
        cout << "Scenario #" << (t + 1) << ":\n";

        if (rayStamps <= 0)
            cout << count;
        else
            cout << "impossible ";
        cout << "\n";
    }
    return 0;
}

File: spoj/3375-STAMPS/solution_3375.py
if __name__ == '__main__':
    for x in range(int(input())):
        total,_ = map(int,input().split(" "))
        friend_stamp  = list(map(int,input().split(" ")))
        friend_stamp.sort(reverse=True)
        count = 0
        for stamps in friend_stamp:
            count += 1
            total -= stamps
            if total <= 0:
                break
        print(f"Scenario #{x+1}:")
        print(("impossible",count)[total <= 0])
